File: articles/history-of-developmental-psychology-from-child-study-to-lifespan-science/r/model_developmental_history.R
#!/usr/bin/env Rscript

suppressPackageStartupMessages({
  library(dplyr)
  library(ggplot2)
  library(tidyr)
})

args <- commandArgs(trailingOnly = FALSE)
script_path <- normalizePath(sub("--file=", "", args[grep("--file=", args)]))
root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
data_path <- file.path(root, "data", "developmental_psychology_history_panel.csv")
outputs_dir <- file.path(root, "outputs")
if (!dir.exists(outputs_dir)) dir.create(outputs_dir, recursive = TRUE)

df <- read.csv(data_path)

paradigm_cols <- c(
  "child_study",
  "maturational",
  "psychoanalytic",
  "behaviorist",
  "cognitive_developmental",
  "sociocultural",
  "attachment_social",
  "ecological",
  "lifespan",
  "developmental_psychopathology",
  "neuroscience_genetics",
  "developmental_systems"
)

selected <- df |>
  filter(year %in% c(1900, 1930, 1950, 1975, 2000, 2025))

write.csv(selected, file.path(outputs_dir, "r_selected_historical_years.csv"), row.names = FALSE)

plot_df <- df |>
  select(year, all_of(paradigm_cols)) |>
  pivot_longer(cols = -year, names_to = "paradigm", values_to = "share")

ggsave(
  filename = file.path(outputs_dir, "r_paradigm_history.png"),
  plot = ggplot(plot_df, aes(x = year, y = share, linetype = paradigm)) +
    geom_line(linewidth = 1) +
    labs(
      title = "Simulated Intellectual History of Developmental Psychology",
      x = "Year",
      y = "Synthetic relative prominence",
      linetype = "Paradigm"
    ) +
    theme_minimal(),
  width = 10,
  height = 6,
  dpi = 160
)

index_df <- df |>
  select(year, child_centered_index, lifespan_index, ecological_systems_index, broadening_index) |>
  pivot_longer(cols = -year, names_to = "index", values_to = "value")

ggsave(
  filename = file.path(outputs_dir, "r_broadening_indexes.png"),
  plot = ggplot(index_df, aes(x = year, y = value, linetype = index)) +
    geom_line(linewidth = 1) +
    labs(
      title = "Historical Broadening Indexes in Developmental Psychology",
      x = "Year",
      y = "Synthetic index",
      linetype = "Index"
    ) +
    theme_minimal(),
  width = 8,
  height = 5,
  dpi = 160
)

peak_summary <- data.frame(
  paradigm = paradigm_cols,
  peak_year = sapply(paradigm_cols, function(col) df$year[which.max(df[[col]])]),
  peak_share = sapply(paradigm_cols, function(col) max(df[[col]]))
)

write.csv(peak_summary, file.path(outputs_dir, "r_paradigm_peak_summary.csv"), row.names = FALSE)
capture.output(summary(df), file = file.path(outputs_dir, "r_history_panel_summary.txt"))

message("Wrote R developmental history outputs.")

File: articles/history-of-developmental-psychology-from-child-study-to-lifespan-science/python/model_developmental_history.py
#!/usr/bin/env python3
"""Model synthetic historical broadening in developmental psychology."""

from __future__ import annotations

from pathlib import Path
import matplotlib.pyplot as plt
import pandas as pd

ROOT = Path(__file__).resolve().parents[1]
DATA = ROOT / "data" / "developmental_psychology_history_panel.csv"
OUT = ROOT / "outputs"
OUT.mkdir(exist_ok=True)

PARADIGMS = [
    "child_study",
    "maturational",
    "psychoanalytic",
    "behaviorist",
    "cognitive_developmental",
    "sociocultural",
    "attachment_social",
    "ecological",
    "lifespan",
    "developmental_psychopathology",
    "neuroscience_genetics",
    "developmental_systems",
]


def main() -> None:
    df = pd.read_csv(DATA)

    selected = df[df["year"].isin([1900, 1930, 1950, 1975, 2000, 2025])]
    selected.to_csv(OUT / "python_selected_historical_years.csv", index=False)

    summary = pd.DataFrame({
        "paradigm": PARADIGMS,
        "peak_year": [int(df.loc[df[col].idxmax(), "year"]) for col in PARADIGMS],
        "peak_share": [float(df[col].max()) for col in PARADIGMS],
        "share_1900": [float(df.loc[df["year"] == 1900, col].iloc[0]) for col in PARADIGMS],
        "share_2025": [float(df.loc[df["year"] == 2025, col].iloc[0]) for col in PARADIGMS],
    })
    summary.to_csv(OUT / "python_paradigm_peak_summary.csv", index=False)

    plt.figure(figsize=(11, 6))
    for col in PARADIGMS:
        plt.plot(df["year"], df[col], label=col)
    plt.xlabel("Year")
    plt.ylabel("Synthetic relative prominence")
    plt.title("Simulated Intellectual History of Developmental Psychology")
    plt.legend(fontsize=7)
    plt.tight_layout()
    plt.savefig(OUT / "python_paradigm_history.png", dpi=160)
    plt.close()

    plt.figure(figsize=(8, 5))
    for col in ["child_centered_index", "lifespan_index", "ecological_systems_index", "broadening_index"]:
        plt.plot(df["year"], df[col], label=col)
    plt.xlabel("Year")
    plt.ylabel("Synthetic index")
    plt.title("Historical Broadening Indexes in Developmental Psychology")
    plt.legend(fontsize=8)
    plt.tight_layout()
    plt.savefig(OUT / "python_broadening_indexes.png", dpi=160)
    plt.close()

    field_forces = df[["year", "institutional_support", "methodological_advantage", "social_relevance", "critique_index"]]
    field_forces.to_csv(OUT / "python_field_forces.csv", index=False)

    with open(OUT / "python_history_model_summary.txt", "w", encoding="utf-8") as f:
        f.write("Synthetic developmental psychology historical model\n")
        f.write("=" * 72 + "\n\n")
        f.write("Selected years\n")
        f.write(selected.to_string(index=False))
        f.write("\n\nParadigm peak summary\n")
        f.write(summary.to_string(index=False))
        f.write("\n\nInterpretive note: values are synthetic teaching data, not empirical bibliometric estimates.\n")

    print("Wrote Python developmental history outputs.")


if __name__ == "__main__":
    main()
